warehouse/integrations/middleware/sql.go

```diff
@@ -0,0 +1,124 @@
+package middleware
+
+import (
+	"context"
+	"database/sql"
+	"time"
+
+	"github.com/rudderlabs/rudder-server/utils/misc"
+
+	"github.com/rudderlabs/rudder-go-kit/logger"
+	"github.com/rudderlabs/rudder-server/warehouse/logfield"
+)
+
+const ExecutingQuery = "executing query"
+
+type Opt func(*DB)
+
+type DB struct {
+	*sql.DB
+
+	since               func(time.Time) time.Duration
+	logger              logger.Logger
+	keysAndValues       []any
+	queryThresholdInSec time.Duration
```

Comment:
[minor] no need to mention `Sec` since it is time.Duration
```suggestion
	slowQueryThreshold time.Duration
```